### DISCUSSION TITLE
Discussion about UI crate

### DISCUSSION THREAD

**Discussion by felipetesc** [zed-industries/zed#9877] [Feature Requests]
At the moment, differently from gpui, the crate UI (zed/crates/ui) is not easily usable for GUI projects, at least not out of the box. 

If we don't add a new package as part of the zed workspace itself, we can't build it easily (gpui.rs website can serve as an example of how straightforward the process is ) I say that because I've successfully created a hello world project as a new crate of zed's workspace using the ui package. 

Though it seems hard right now given the amount of work and limited resources, in the future some changes could prove invaluable to facilitate the process of creating new GUIs with Zed's ui infrastructure. Maybe, but not necessarily, to adjust UI crate in order to call it directly, without the need of gpui(gpui as I read is part of the ui crate), could prove interesting since new releases of Zed GUI would need only the zed/crates/ui, and at the same time we could use, for external projects, only the UI crate.

So, I started this discussion, to start thinking about UI crate as a standalone usable crate.   

Honestly, the work done so far is incredible. I hope I didn't make too many mistakes about what I wrote, and that this discussion is positive and serves to create an even better project.

Thanks.

**Comment by maxdeviant** [comment]
Right now the `ui` crate is intended specifically for use with Zed.

At some point we may end up extracting a more general-purpose library containing various components for GPUI, but we don't currently have any concrete plans for `ui` to be that.
